lib/systems/include/SysClock.h

```diff
@@ -0,0 +1,42 @@
+#ifndef __SYSCLK_H__
+#define __SYSCLK_H__
+
+struct TriggerBits_s
+{
+    bool trigger1000:1;
+    bool trigger500:1;
+    bool trigger100:1;
+    bool trigger50:1;
+    bool trigger10:1;
+    bool trigger5:1;
+    bool trigger1:1;
+};
+
+struct SysTick_s
+{
+    long millis;
+    long micros;
+    TriggerBits_s triggers;
+};
+
+class SysClock
+{
+private:
+    enum TriggerIndices_s
+    {
+        TRIG_1000 = 0,
+        TRIG_500 = 1,
+        TRIG_100 = 2,
+        TRIG_50 = 3,
+        TRIG_10 = 4,
+        TRIG_5 = 5,
+        TRIG_1 = 6,
+        NUM_TRIGGERS = 7
+    };
+    unsigned long triggerTimes[NUM_TRIGGERS];
```

Comment:
I may be missing something here, but why use an enum for the trigger indexes? What would be the hurt in just updating  a member instance of the trigger bits? also if you want to refer to an example implementation of how to do a system clock, std::chrono is a good place to reference because this to me looks like more of a task manager than a system clock:

https://github.com/gcc-mirror/gcc/blob/master/libstdc++-v3/src/c++11/chrono.cc#L52

Reply:
the enum is used as internal macros to avoid magic numbers. I'm actually surprised it's not complaining that I'm not `static_cast`ing uses of the enum.

I could rename this to SysTicker or something, SysClock is a misnomer. It does what I expect though. Are you proposing functional changes?

Reply:
At this point we should only need to make changes to code if it violates layer separation or is wrong functionality-wise. This looks simple enough and easy-to-use to me. If it works properly there's no need to change the implementation. My only question is would overflow ever be an issue if we kept recalculating by adding numbers (in theory millis() and micros() also return unsigned long so I might just be stupid).